yesod-test/Yesod/Test.hs

```diff
@@ -276,6 +277,24 @@ yesodSpecWithSiteGenerator getSiteAction yspecs =
             , yedResponse = Nothing
             }
 
+yesodSpecWithSiteGenerator' :: YesodDispatch site
```

Comment:
What's the purpose of this function? Why would I use this function instead of `yesodSpecWithSiteGenerator`? We definitely need a Haddock for this. (Also, when written, we need a `@since` comment, ChangeLog entry, and cabal file version bump.)